### PR TITLE
YYA three month followup

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

File: drivers/ma_yya_followup_report/app/models/ma_yya_followup_report/report.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+module MaYyaFollowupReport
+  class Report
+    include ArelHelper
+    include Filter::FilterScopes
+
+    def initialize(filter)
+      @filter = filter
+      @end_date = @filter.on
+      @start_date = @end_date - 2.months - 3.weeks
```
Comment (eanders) [inline drivers/ma_yya_followup_report/app/models/ma_yya_followup_report/report.rb]: Should this be?
```suggestion
      @start_date = @end_date - 3.months + 1.weeks
```

File: drivers/ma_yya_followup_report/app/models/ma_yya_followup_report/report.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    def columns
+      window = ::Arel::Nodes::Window.new.partition(c_t[:id])
+      {
+        id: :id,
+        first_name: :FirstName,
+        last_name: :LastName,
+        engagement_date: she_t[:first_date_in_program].minimum.over(window),
+        last_seen: cls_t[:InformationDate].maximum.over(window),
+      }
```
Comment (eanders) [inline drivers/ma_yya_followup_report/app/models/ma_yya_followup_report/report.rb:40]: a new toy?

File: drivers/ma_yya_followup_report/app/models/ma_yya_followup_report/report.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        distinct.
+        joins(service_history_enrollments: [enrollment: :current_living_situations]).
+        merge(enrollment_scope).
+        merge(contact_scope)
+    end
+
+    private def enrollment_scope
+      scope = ::GrdaWarehouse::ServiceHistoryEnrollment.
+        entry.
+        open_between(start_date: @start_date, end_date: @end_date)
```
Comment (eanders) [inline drivers/ma_yya_followup_report/app/models/ma_yya_followup_report/report.rb]: Any reason not to use:
```suggestion
      scope = ::GrdaWarehouse::ServiceHistoryEnrollment.entry
      scope = filter_for_range(scope)
```

File: drivers/ma_yya_followup_report/app/models/ma_yya_followup_report/report.rb [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      filter_for_age(scope)
+    end
+
+    private def contact_scope
+      ::GrdaWarehouse::Hud::CurrentLivingSituation.
+        between(start_date: @start_date, end_date: @end_date)
+    end
+
+    def self.yya_projects(user)
+      ::GrdaWarehouse::Hud::Project.options_for_select(user: user)
```
Comment (eanders) [inline drivers/ma_yya_followup_report/app/models/ma_yya_followup_report/report.rb]:
```suggestion
      filter.project_options_for_select(user: user)
```

File: drivers/ma_yya_followup_report/app/models/ma_yya_followup_report/report.rb (hunk 2 above)
Comment (awisema) [reply drivers/ma_yya_followup_report/app/models/ma_yya_followup_report/report.rb:40]: Yep -- Rails 6 support for window functions... Line 30 is a little ugly but it produces `MAX("CurrentLivingSituation"."InformationDate") OVER (PARTITION BY "Client"."id")` which is nice!

**Pull request merged**